features/enlinkd/service/api/src/main/java/org/opennms/netmgt/enlinkd/service/api/BridgeSimpleConnection.java

```diff
@@ -148,106 +149,120 @@ private BridgeSimpleConnection(BridgeForwardingTable xBridge,
-        if (LOG.isDebugEnabled()) {
-            LOG.debug("bridge: [{}] <-> [{}] common (learned mac): {}",
-                  m_yBridge.getNodeId(),
-                  m_xBridge.getNodeId(),
-                  commonlearnedmacs);
-        }
+
+        Set<String> commonlearnedmacs = new HashSet<>(m_xBridge.getMactoport().keySet());
+        commonlearnedmacs.retainAll(new HashSet<>(m_yBridge.getMactoport().keySet()));
 
         if (m_yxPort != null && m_xyPort == null) { 
-            try {
-                m_xyPort = condition2(commonlearnedmacs,
-                                             m_yxPort,
-                                             m_yBridge,
-                                             m_xBridge);
-            } catch (BridgeTopologyException e) {
-                LOG.warn("bridge: [{}] -> [{}], {}",
-                          m_xBridge.getNodeId(),
-                          m_yBridge.getNodeId(),
-                          e.getMessage());
-                m_xyPort = conditionB(commonlearnedmacs, m_xBridge);
-            }
-            return;
-        } 
+            m_xyPort = condition2(
+                            commonlearnedmacs,
+                            m_yxPort,
+                            m_yBridge,
+                            m_xBridge);
+        }
         
         if (m_yxPort == null && m_xyPort != null) {
-            try {
-                m_yxPort = BridgeSimpleConnection.condition2(commonlearnedmacs,
-                                             m_xyPort,
-                                             m_xBridge,
-                                             m_yBridge);
-            } catch (BridgeTopologyException e) {
-                LOG.warn("bridge: [{}] -> [{}], {}",
-                          m_yBridge.getNodeId(),
-                          m_xBridge.getNodeId(),
-                          e.getMessage());
-                m_yxPort = conditionB(commonlearnedmacs, m_yBridge);
-            }
+            m_yxPort = BridgeSimpleConnection.condition2(
+                            commonlearnedmacs,
+                            m_xyPort,
+                            m_xBridge,
+                            m_yBridge);
+        }
+
+        if (m_xyPort != null && m_yxPort != null) {
             return;
-        } 
-        
-        List<BridgePort> ports = BridgeSimpleConnection.condition3(commonlearnedmacs,
-               m_xBridge,
-               m_yBridge);
-        m_xyPort = ports.get(0);
-        m_yxPort= ports.get(1);
+        }
+
+        if (m_xyPort == null && m_yxPort == null) {
+            List<BridgePort> ports = BridgeSimpleConnection.condition3(commonlearnedmacs, m_xBridge, m_yBridge);
+            if (ports.size() == 2) {
+                m_xyPort = ports.get(0);
+                m_yxPort = ports.get(1);
+                return;
+            }
+        }
+
+        if (m_xyPort == null && m_yxPort == null) {
```

Comment:
Sam condition like line #208. Can be combined together, right?

Reply:
commit c0b86a54b8980714d18ad0ea2348ff0eda3da5ef
Author: Antonio <rssntn67@yahoo.it>
Date:   Mon May 23 22:34:52 2022 +0200

    Two BridgePort Node - Topology Mismatch

    According to pape Christian put together
    See convertion on PR 4738